cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/store/AppMetadataStore.java

```diff
@@ -2567,12 +2576,10 @@ private void writeApplicationSerialized(String namespaceId, String appId, String
           change.getDescription()));
     }
     fields.add(Fields.booleanField(StoreDefinition.AppMetadataStore.LATEST_FIELD, markAsLatest));
-
-    if (sourceControlMeta != null) {
-      fields.add(Fields.stringField(StoreDefinition.AppMetadataStore.SOURCE_CONTROL_META,
-          GSON.toJson(sourceControlMeta)));
-    }
     getApplicationSpecificationTable().upsert(fields);
+    getSourceControlMetadataStore().setAppSourceControlMeta(
```

Comment:
Would we have a write operation to scm metadata every time app upserts?

Reply:
Write operation is performed every time we do deploy (including after pull operation). 
If we just do deploy (without pull operation), a new version of application is created, which is not connected to the git hence sourcecontrolmetadata is null in this case and needs to be updated in the source_control_metadata table. This table contains scm metadata of only latest versions of entities (eg application).